src/charm.py

```diff
@@ -205,6 +187,74 @@ def _check_kf_profiles(self, interfaces):
 
         return kf_profiles
 
+    def _create_resources(self, resource_type: List[str] = None) -> None:
+        """Creates the resources for the charm."""
+        if resource_type is None:
+            resource_type = self._resource_files.keys()
+        for resource_type in resource_type:
+            resource_file = self._resource_files[resource_type]
+            manifest = self.env.get_template(resource_file).render(self._context)
+            for obj in codecs.load_all_yaml(manifest):
+                try:
+                    self.lightkube_client.create(obj)
+                except ApiError as e:
+                    if e.status.code == 409:
+                        self.logger.info("replacing resource: %s.", str(obj.to_dict()))
+                        self.logger.debug(f"manifest is {manifest}")
+                        try:
+                            self.lightkube_client.patch(
+                                type(obj),
+                                obj.metadata.name,
+                                obj,
+                                patch_type=PatchType.MERGE,
+                            )
+                        except ApiError as e:
+                            if e.status.code == 409:
+                                self.logger.info(
+                                    "Unable to replace resource: %s. Skipping.",
+                                    str(obj.to_dict()),
+                                )
+                            else:
+                                raise e
+                    else:
+                        self.logger.debug(
+                            "failed to create resource: %s.", str(obj.to_dict())
+                        )
+                        raise e
+
+    def main(self, event) -> None:
+        """Main entry point for the Charm."""
+        try:
+            self._check_container_connection()
+            self._check_model_name()
+            self._check_leader()
+            interfaces = self._get_interfaces()
+            kf_profiles = self._check_kf_profiles(interfaces)
+        except CheckFailed as e:
+            self.model.unit.status = e.status
+            return
+        try:
+            self.unit.status = MaintenanceStatus("Creating k8s resources")
+            try:
```

Comment:
I cant use chisme because it does not implement the patch on apply failure .... sometimes the clusterrolebinding exists before so i need to patch it ... or i was expecting the configmap can be rewritten but now it cant so ..

Reply:
So if I understand this conversation correctly, the main reason we are not using chisme here is due to the `dashboard-configmap` option. We don't want to step on users' toes if they are using that option and pointing to their own config map. I believe this is a bit of an antipattern that we usually discourage in other operators, once we encourage users to "bring their own config" it becomes very hard to maintain the different use cases that users can come up with. Considering also how we are going to overwrite things with the sidebar relation this just becomes a lot more complex than necessary. I believe we should go ahead and remove the configmap option entirely and rely on the operator to be the only one that can create it. This will make our lives a lot easier.